lib/taskana-simplehistory-provider/src/test/java/acceptance/historyquery/QueryHistoryAccTest.java

```diff
@@ -0,0 +1,302 @@
+package acceptance.historyquery;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.time.Instant;
+import java.util.List;
+
+import org.junit.Test;
+
+import acceptance.AbstractAccTest;
+import pro.taskana.BaseQuery.SortDirection;
+import pro.taskana.TimeInterval;
+import pro.taskana.simplehistory.impl.HistoryEventImpl;
+import pro.taskana.simplehistory.impl.SimpleHistoryServiceImpl;
+import pro.taskana.simplehistory.query.HistoryQuery;
+import pro.taskana.simplehistory.query.HistoryQueryColumnName;
+
+public class QueryHistoryAccTest extends AbstractAccTest {
+
+    private SimpleHistoryServiceImpl historyService;
+
+    public QueryHistoryAccTest() {
+        super();
+        this.historyService = (SimpleHistoryServiceImpl) taskanaHistoryEngine.getTaskanaHistoryService();
+    }
+
+    @Test
+    public void testListValuesAscendingAndDescending() {
+        List<String> defaultList = historyService.createHistoryQuery().listValues(HistoryQueryColumnName.COMMENT, null);
+        List<String> ascendingList = historyService.createHistoryQuery().listValues(HistoryQueryColumnName.COMMENT, SortDirection.ASCENDING);
+        List<String> descendingList = historyService.createHistoryQuery().listValues(HistoryQueryColumnName.COMMENT, SortDirection.DESCENDING);
+
+        assertEquals(3, ascendingList.size());
+        assertArrayEquals(defaultList.toArray(), ascendingList.toArray());
+        assertEquals(ascendingList.get(2), descendingList.get(0)); //Last element in ascendingList should be the same as first in the descendingList
+    }
+
+    @Test
+    public void testComplexQuery() {
+        HistoryQuery query = historyService.createHistoryQuery()
+            .businessProcessIdLike("just some string", "BPI:%") // contains all values
```

Comment:
I would recommend to remove all this kind of comments, code should be explained by it self without comments.

Reply:
The comments are more about the test data than the code.
I personally find it annoying to look up all Values in the testdata by hand to understand the query, but I'm okay with deleting the comments.

Reply:
I prefer no comments in the code, at least if it is not strictly necessary.

Reply:
ok!